packages/sit-onyx/src/components/OnyxButton/OnyxButton.vue

```diff
@@ -0,0 +1,157 @@
+<script lang="ts" setup>
+import type { ButtonProps } from "./types";
+const props = withDefaults(defineProps<ButtonProps>(), {
+  label: "",
+  isLoading: false,
+  isDisabled: false,
+  type: "button",
+  color: "primary",
+  variant: "default",
+});
+
+const emit = defineEmits<{
+  /**
+   * Emitted when the button is clicked
+   */
+  click: [];
+}>();
+</script>
+
+<template>
+  <div
+    tabindex="0"
+    class="onyx-button"
+    :class="{
+      [`onyx-button--${props.color}`]: true,
+      [`onyx-button--${props.variant}--${props.color}`]: true,
+    }"
```

Comment:
```suggestion
    :class="[`onyx-button--${props.color}`, `onyx-button--${props.variant}`]"
```

We already have a class with props.color in the line above so we don't need to redundantly add it to the variant class.

In CSS we can just combine the selectors, for example:
```css
.onyx-button {
  &--primary&--default {
    // matches a button with both color primary and variant default
  }  
}
```

Reply:
![image](https://github.com/SchwarzIT/onyx/assets/60856270/ca2f1983-96cb-45c3-95bb-2c33f15232b2)

Reply:
Hmm than its probably
```css
.onyx-button {
  &--primary.onyx-button--default {
    // matches a button with both color primary and variant default
  }  
}
```

My main point is to not have two redundant classes :) I'd prefer to just put the classes on the button for what the button is so e.g. "default" and "primary" and then style it accordingly. Instead of having classes for every combination like "--default--primary", "--default--secondary" etc.

Reply:
The second one works!! Thanks :)